src/main/io/gps.c

```diff
@@ -573,110 +755,393 @@ static void ubloxSetMessageRate(uint8_t messageClass, uint8_t messageID, uint8_t
+                    // print the init string for the baudrate we want to be at
+                    serialPrint(gpsPort, "$PSRF100,1,115200,8,1,0*05\r\n");
+                    gpsData.state_position++;
+                    break;
+                default: 
+                    // we're now (hopefully) at the correct rate, next state should switch to it
+                    // TO DO: Check that this actually works, it seems broken at present
+                    gpsData.state_position = 0;
+                    gpsSetState(GPS_STATE_CHANGE_BAUD);
+                    break;
+            }
             break;
+#endif
+        case GPS_STATE_CHANGE_BAUD:
+#if !defined(GPS_NMEA_TX_ONLY)
+            // wait a short time between sending commands
+            // note that no commands are sent to request the packets we need
+            now = millis();
+            if (cmp32(now, gpsData.state_ts) < 500) {
+                return;
+            }
+            gpsData.state_ts = now;
+
+            if (gpsData.state_position < 1) { // first run after boot up 
+                serialSetBaudRate(gpsPort, baudRates[gpsInitData[gpsData.baudrateIndex].baudrateIndex]);
+                gpsData.state_position++;
+            } else if (gpsData.state_position < 2) {
+                // *** this message also appears to fail ***//
+                // NMEA reports back at whatever speed the module is configured to send at, not 5Hz
+                serialPrint(gpsPort, "$PSRF103,00,6,00,0*23\r\n"); // set GGA rate to 5Hz
+                gpsData.state_position++;
+            } else if (gpsData.state_position < 3) {
+                // special initialization for NMEA ATGM336 and similar GPS recivers - should be done only once
+                if (!atgmRestartDone) {
+                    atgmRestartDone = true;
+                    serialPrint(gpsPort, "$PCAS02,100*1E\r\n");  // 10Hz refresh rate
+                    serialPrint(gpsPort, "$PCAS10,0*1C\r\n");    // hot restart
+                }
+                gpsData.state_position++;
+            } else if (gpsData.state_position < 4) {
+                // try to disable UBX NAV-PVT and NAV-SAT messages via ValSet
+                ubloxSetMessageRateValSet(CFG_MSGOUT_UBX_NAV_PVT_UART1, 0);
+                ubloxSetMessageRateValSet(CFG_MSGOUT_UBX_NAV_SAT_UART1, 0);
+                gpsData.state_position++;
+            } else if (gpsData.state_position < 5) {
+                // try to disable UBX NAV-PVT and NAV-SAT messages via UBX
+                ubloxSetMessageRate(CLASS_NAV, MSG_NAV_PVT, 0);
+                ubloxSetMessageRate(CLASS_NAV, MSG_NAV_SAT, 0);
+                ubloxSetMessageRate(CLASS_NAV, MSG_NAV_SVINFO, 0);
+                gpsData.state_position++;
+            } else if (gpsData.state_position < 6) {
+                if (!(isConfiguratorConnected())) {
+                    // disable GSV MESSAGES
+                    // *** THIS COMMAND FAILS TO DISABLE GSV MESSAGES WHEN SENT ****
```

Comment:
I wasn't sure if ublox GPS module is always processing both binary and text commands. It may be reason why text command fails

Reply:
Thanks for explanation

Reply:
The code only parses one of NMEA or UBX depending on the `gpsConfig()->provider` (cli `gps_provider`) setting.
If set to UBX, an NMEA message will not get recognized because the parser is only looking for the UBX preamble, so the NMEA message bytes will get tossed until it syncs up with a UBX message.

Using only NMEA commands, we could turn off the reception of UBX commands when we set the module UART config.
For example, we can send `$PUBX,41,1,0003,0002,115200,0*1E\r\n`, where:
- the first `0003` is the enabled module input protocols, byte 0 is UBX and byte 1 is NMEA, so listening for both;
- the second `0002` is the enabled module output protocols, same bytes as above, so only sending NMEA.
This is a "big stick" approach to disabling all outbound UBX message from the module, but maybe easier than trying to turn off each UBX message if we really want to receive no UBX messages at all?

The UBX code does the opposite, although ironically it uses an NMEA command to do so
There is UBX-CFG-PRT port config command and newer VALSET key/values to do it using UBX only.
It sends `$PUBX,41,1,0003,0001,115200,0*1E\r\n`, where:
- `0003` for listening for both protocols from us;
- `0001` for only sending UBX protocol to us.
Theoretically, if that works as advertised, the steps we do in config to turn off each NMEA command shouldn't even be needed, since we've completely turned off NMEA incoming messages at the protocol level.

Totally agree, cleaning up all this is for another PR...